docs: Improve documentation (#289)
ansys/pre-commit-hooks: doc/source/getting-started/add-license-headers/index.rst

``add-license-headers`` setup
=============================

The ``add-license-headers`` hook adds and updates license headers in specified directories and
file types using `REUSE <https://reuse.software/>`_. Here is a sample license header for a
Python file generated by the template with a start year of 2023:

.. image:: /_static/images/license-header-light.png
   :class: only-light
   :alt: Picture of Python file's license header in light mode
   :width: 480px
   :align: center

.. image:: /_static/images/license-header-dark.PNG
   :class: only-dark
   :alt: Picture of Python file's license header in dark mode
   :width: 480px
   :align: center

To get started, add the hook to your ``.pre-commit-config.yaml`` file:

.. code:: yaml

  - repo: https://github.com/ansys/pre-commit-hooks
    rev: v0.5.1
    hooks:
    - id: add-license-headers

> The hook runs on protocol buffer files in any directory, as well as Python files within directories named

Reviewer: Earlier PROTO files was used. Maybe, if this is the first reference, do this: The hook runs on protocol buffer (PROTO) files in any directory, ....". Then just use PROTO files as was done.

``src``, ``examples``, and ``tests``.

If your repository was created before the current year, add the ``--start_year`` argument with the

> year your first file was committed. The ``ansys-pre-commit-hooks`` repository started in 2023,

Reviewer: ```suggestion
year that your first file was committed. The ``ansys-pre-commit-hooks`` repository started in 2023,
```

so the ``.pre-commit-config.yaml`` file looks like this:

.. code:: yaml

  - repo: https://github.com/ansys/pre-commit-hooks
    rev: v0.5.1
    hooks:
    - id: add-license-headers
      args:
      - --start_year=2023

By default, the
`ansys.jinja2 <https://github.com/ansys/pre-commit-hooks/blob/main/src/ansys/pre_commit_hooks/assets/.reuse/templates/ansys.jinja2>`_
template is used to generate the license headers for files.

The template contains the following variables:

> - ``{{ copyright_line }}``: By default, contains the current year and the phrase "ANSYS, Inc. and/or its affiliates."
> - ``{{ expression }}``: The SPDX-License-Identifier expression, which is "MIT" by default.

Reviewer: ```suggestion
- ``{{ copyright_line }}``: Contains the current year and the phrase "ANSYS, Inc. and/or its affiliates." by default.
- ``{{ expression }}``: Contains the SPDX-License-Identifier expression, which is "MIT" by default.
```

``add-license-headers`` hook arguments
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

.. list-table::
    :header-rows: 1
    :widths: 20 20 60
    :width: 100%

    * - Argument
      - Default value
      - Description
    * - ``--custom_copyright``
      - ``ANSYS, Inc. and/or its affiliates.``

>       - The copyright line you want to include in the license header.
>     * - ``--custom_template``
>       - ``ansys``
>       - The name of the ``.jinja2`` template file located in the ``.reuse/templates/`` directory.
>     * - ``--custom_license``
>       - ``MIT``
>       - The SPDX-License-Identifier expression you want to include in the license header. To view
>         the list of valid SPDX license expressions, see the
>         `SPDX License List <https://github.com/spdx/license-list-data/tree/main/text>`_.
>     * - ``--start_year``
>       - ``2023``
>       - The year the first file was committed to the repository.
>     * - ``--ignore_license_check``
>       - ``False``

Reviewer: ```suggestion
      - Copyright line to include in the license header.
    * - ``--custom_template``
      - ``ansys``
      - Name of the ``.jinja2`` template file located in the ``.reuse/templates/`` directory.
    * - ``--custom_license``
      - ``MIT``
      - SPDX-License-Identifier expression to include in the license header. To view
        the list of valid SPDX license expressions, see the
        `SPDX License List <https://github.com/spdx/license-list-data/tree/main/text>`_.
    * - ``--start_year``
      - ``2023``
      - Year that the first file was committed to the repository.
    * - ``--ignore_license_check``
      - ``False``
```
We ususally avoid using an article (The, A, An) as the first word in a brief description. We also avoid words like "you want" as they are implied.

>       - Whether or not to check for the license in the header.

Reviewer: ```suggestion
      - Whether to check for the license in the header.
```
Some repos use a description like this when the data type of Boolean isn't given: "Boolean indicating whether to check for the license in the header."
